packages/jni-swig-stub/src/main/jni/realm_api_helpers.cpp

```diff
@@ -235,24 +231,37 @@ class CustomJVMScheduler : public realm::util::Scheduler {
 //
 // I suspect this could be related to the fact that jni.h defines jlong differently between Android (typedef int64_t)
 // and JVM which is a (typedef long long) resulting in a different signature of the method that could be found by the linker.
-void invoke_core_notify_callback(int64_t core_notify_function) {
-    auto notify = reinterpret_cast<std::function<void()> *>(core_notify_function);
-    (*notify)();
+void invoke_core_notify_callback(int64_t scheduler) {
+    realm_scheduler_perform_work(reinterpret_cast<realm_scheduler_t *>(scheduler));
 }
 
-// TODO refactor to use public C-API https://github.com/realm/realm-kotlin/issues/496
 realm_t *open_realm_with_scheduler(int64_t config_ptr, jobject dispatchScheduler) {
-    auto *cfg = reinterpret_cast<realm_config_t * >(config_ptr);
+    auto config = reinterpret_cast<realm_config_t *>(config_ptr);
     // copy construct to not set the scheduler on the original Conf which could be used
-    // to open Frozen Realm for instance.
-    auto copyConf = *cfg;
+    // to open Frozen Realm for instance. realm_clone doesn't produce a copy but just increases the
```

Comment:
The constraint you mention is regarding Realm instances not `realm_config` why can't we use? 
```C++
    realm_config_t * config_clone = static_cast<realm_config_t *>(realm_clone(config));

    if (dispatchScheduler) {
        auto jvmScheduler = new CustomJVMScheduler(dispatchScheduler);
        auto scheduler = realm_scheduler_new(
                jvmScheduler,
                [](void *userdata) { delete(static_cast<CustomJVMScheduler *>(userdata)); },
                [](void *userdata) { static_cast<CustomJVMScheduler *>(userdata)->notify(); },
                [](void *userdata) { return static_cast<CustomJVMScheduler *>(userdata)->is_on_thread(); },
                [](const void *userdata, const void *userdata_other) { return userdata == userdata_other; },
                [](void *userdata) { return static_cast<CustomJVMScheduler *>(userdata)->can_invoke(); }
        );
        jvmScheduler->set_scheduler(scheduler);
        realm_config_set_scheduler(config_clone, scheduler);
```

Reply:
It fails when I try to `realm_clone` it
```
signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x138
Cause: null pointer dereference
    rax 0000000000000138  rbx 000072b77f746c00  rcx 0000000000000001  rdx 000072b80f9c15a8
    r8  0000000000000001  r9  000072b77f76f9c8  r10 0000000000000000  r11 0000000000000206
    r12 000072b733c776b0  r13 000072b734d881e6  r14 000000000000000c  r15 000072b733c77ff8
    rdi 0000000000000138  rsi 0000000000000138
    rbp 000072b733c77370  rsp 000072b733c771e0  rip 000072b732c9f3c8
backtrace:
      #00 pc 00000000007153c8  /data/app/io.realm.testapp.test-TRZkcTCmYc26qjbEBVRgOA==/lib/x86_64/librealmc.so (_ZNSt6__ndk14swapIPN5realm4util9SchedulerEEENS_9enable_ifIXaasr21is_move_constructibleIT_EE5valuesr18is_move_assignableIS6_EE5valueEvE4typeERS6_S9_+56) (BuildId: 6fc6aa1034c801b34562a342ebf51f4b08fb0c46)
      #01 pc 0000000000715370  /data/app/io.realm.testapp.test-TRZkcTCmYc26qjbEBVRgOA==/lib/x86_64/librealmc.so (std::__ndk1::shared_ptr<realm::util::Scheduler>::swap(std::__ndk1::shared_ptr<realm::util::Scheduler>&)+32) (BuildId: 6fc6aa1034c801b34562a342ebf51f4b08fb0c46)
      #02 pc 000000000070cb96  /data/app/io.realm.testapp.test-TRZkcTCmYc26qjbEBVRgOA==/lib/x86_64/librealmc.so (std::__ndk1::shared_ptr<realm::util::Scheduler>::operator=(std::__ndk1::shared_ptr<realm::util::Scheduler> const&)+70) (BuildId: 6fc6aa1034c801b34562a342ebf51f4b08fb0c46)
      #03 pc 000000000070cb3b  /data/app/io.realm.testapp.test-TRZkcTCmYc26qjbEBVRgOA==/lib/x86_64/librealmc.so (realm_config_set_scheduler+43) (BuildId: 6fc6aa1034c801b34562a342ebf51f4b08fb0c46)
      #04 pc 00000000006ecfde  /data/app/io.realm.testapp.test-TRZkcTCmYc26qjbEBVRgOA==/lib/x86_64/librealmc.so (open_realm_with_scheduler(long, _jobject*)+430) (BuildId: 6fc6aa1034c801b34562a342ebf51f4b08fb0c46)
```
But a bit weird ... because it shouldn't even be cloneable: https://github.com/realm/realm-core/blob/master/test/object-store/c_api/c_api.cpp#L324 
🤔

Reply:
Added the stack trace an outcome of trying it out to #496, but will leave as is for now.